src/render/ItemRenderer.ts

```diff
@@ -48,7 +48,16 @@ export class ItemRenderer extends Renderer {
 		if (!tint && this.item.id.namespace === Identifier.DEFAULT_NAMESPACE) {
 			tint = getItemColor(this.item)
 		}
-		const mesh = model.getDisplayMesh('gui', this.resources, tint)
+		var additionalMesh = undefined
+		if (SpecialRenderers.has(this.item.id.toString())){
+			additionalMesh = SpecialRenderer[this.item.id.toString()]({}, this.resources, Cull.none())
+			// undo the scaling done by the special renderer
+			const t = mat4.create()
+			mat4.identity(t)
+			mat4.scale(t, t, [16, 16, 16])
+			additionalMesh.transform(t)
+		}
+		const mesh = model.getDisplayMesh('gui', this.resources, tint, additionalMesh)
```

Comment:
Can the `additionalMesh` not be merged with the mesh returned from `getDisplayMesh` instead of passing it to the function?
```ts
const mesh = model.getDisplayMesh('gui', this.resources, tint)
mesh.merge(additionalMesh)
```

Reply:
The `additionalMesh` also need to get all the transformations that `getDisplayMesh` applies on the mesh, so it has to be merged before the transformations (or be transformed separately).